tools/testsys/src/run.rs

```diff
@@ -101,16 +83,65 @@ pub(crate) struct Run {
     migration_target_version: Option<String>,
 }
 
+/// This is a CLI parsable version of `testsys_config::GenericVariantConfig`.
+#[derive(Debug, Parser)]
+struct CliConfig {
+    /// The repo containing images necessary for conformance testing. It may be omitted to use the
+    /// default conformance image registry.
+    #[clap(long, env = "TESTSYS_CONFORMANCE_REGISTRY")]
+    conformance_registry: Option<String>,
+
+    /// The name of the cluster for resource agents (EKS resource agent, ECS resource agent). Note:
+    /// This is not the name of the `testsys cluster` this is the name of the cluster that tests
+    /// should be run on. If no cluster name is provided, the bottlerocket cluster
+    /// naming convention `{{arch}}-{{variant}}` will be used.
+    #[clap(long, env = "TESTSYS_TARGET_CLUSTER_NAME")]
+    target_cluster_name: Option<String>,
+
+    /// The image that should be used for conformance testing. It may be omitted to use the default
+    /// testing image.
+    #[clap(long, env = "TESTSYS_CONFORMANCE_IMAGE")]
+    conformance_image: Option<String>,
+
+    /// The role that should be assumed by the agents
+    #[clap(long, env = "TESTSYS_ASSUME_ROLE")]
+    assume_role: Option<String>,
+
+    /// Specify the instance type that should be used. This is only applicable for aws-* variants.
+    /// It can be omitted for non-aws variants and can be omitted to use default instance types.
+    #[clap(long, env = "TESTSYS_INSTANCE_TYPE")]
+    instance_type: Option<String>,
+
+    /// Add secrets to the testsys agents (`--secret aws-credentials=my-secret`)
+    #[clap(long, short, parse(try_from_str = parse_key_val), number_of_values = 1)]
+    secret: Vec<(String, SecretName)>,
+}
+
+impl From<CliConfig> for GenericVariantConfig {
+    fn from(val: CliConfig) -> Self {
+        GenericVariantConfig {
+            cluster_names: val.target_cluster_name.into_iter().collect(),
+            instance_type: val.instance_type,
+            secrets: val.secret.into_iter().collect(),
+            agent_role: val.assume_role,
+            conformance_image: val.conformance_image,
+            conformance_registry: val.conformance_registry,
+        }
+    }
+}
+
 impl Run {
     pub(crate) async fn run(self, client: TestManager) -> Result<()> {
         let variant =
             Variant::new(&self.variant).context("The provided variant cannot be interpreted.")?;
         debug!("Using variant '{}'", variant);
-        let secrets = if self.secret.is_empty() {
-            None
-        } else {
-            Some(self.secret.into_iter().collect())
-        };
+
+        // If a lock file exists, use that, otherwise use Test.toml or default
+        let test_config = TestConfig::from_path_or_lock(&self.test_config_path, true)
+            .context("Unable to read test config")?;
+
+        let test = test_config.test.as_ref().cloned().unwrap_or_default();
```

Comment:
nit: I'd prefer a more specific variable name, like
```suggestion
        let test_opts = test_config.test.as_ref().cloned().unwrap_or_default();
```